### PR TITLE
feature-261: Role get, list, update

### PR DESCRIPTION
**Pull request by atye**
# Description

This PR adds functionality for role list, get and update in the role-service. 

The `role get` command is updated with a `name` flag. Previously, it worked like `karavictl role get roleName`. The flag is being added to align with how other commands work.

How it works:

## role get
1. karavictl builts role client and makes Get request
2. role-service queries the `common` configMap in k8s to get configured roles
3. role-service filters the configured roles for the requested name
4. role-service marshals the filtered roles and returns a []byte response
5. karavictl receives the response and formats a json output

## role list
1. karavictl builts role client and makes List request
2. role-service queries the `common` configMap in k8s to get configured roles
3. role-service marshals the filtered roles and returns a []byte response
4. karavictl receives the response and formats a json output

## role update
1. karavictl builts role client and makes Update request
2. role-service queries the `common` configMap in k8s to get configured roles
5. role-service validates the updated role information
6. role-service replaces the relevant role with the new role
7. role-service updates the `common` configMap in k8s to persist the update

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/261|

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

This is tested with unit tests and manual verification that role get, list, and update work.

### CODE REVIEW

File: cmd/karavictl/cmd/role_update_grpc_test.go [hunk 1]
```diff
@@ -0,0 +1,130 @@
+// Copyright © 2021 Dell Inc., or its subsidiaries. All Rights Reserved.
```
Comment (sharmilarama) [inline cmd/karavictl/cmd/role_update_grpc_test.go]: Change to 2022

File: cmd/karavictl/cmd/role.go [hunk 2]
```diff
@@ -68,7 +68,7 @@ func NewRoleCmd() *cobra.Command {
 	}
 
 	roleCmd.PersistentFlags().String("addr", "", "address of the csm-authorzation role service")
```
[the remark this answers is not in the document]
Comment (atye) [reply cmd/karavictl/cmd/role.go]: Fixed.

File: cmd/karavictl/cmd/role_update_grpc_test.go (hunk 1 above)
Comment (atye) [reply cmd/karavictl/cmd/role_update_grpc_test.go]: Fixed.

**Review by hoppea2 — APPROVED** [review]

**Review by hoppea2 — COMMENTED** [review]
LGTM - Has the role creation and get/list/update been tested in the rpm as well?

**Comment by atye** [comment]
> LGTM - Has the role creation and get/list/update been tested in the rpm as well?

Yes, rpm functionality works. e2e will be run as well to verify.

**Review by alikdell — APPROVED** [review]

**Comment by atye** [comment]
run e2e test

**Pull request merged**